Mathlib/MeasureTheory/Integral/Lebesgue.lean

```diff
@@ -1432,6 +1432,11 @@ theorem lintegral_add_compl (f : α → ℝ≥0∞) {A : Set α} (hA : Measurabl
   rw [← lintegral_add_measure, Measure.restrict_add_restrict_compl hA]
 #align measure_theory.lintegral_add_compl MeasureTheory.lintegral_add_compl
 
+theorem setLintegral_compl {f : α → ℝ≥0∞} {s : Set α} (hsm : MeasurableSet s)
+    (hfs : ∫⁻ x in s, f x ∂μ ≠ ∞) :
+    ∫⁻ x in sᶜ, f x ∂μ = ∫⁻ x, f x ∂μ - ∫⁻ x in s, f x ∂μ := by
+  rw [← lintegral_add_compl (μ := μ) f hsm, ENNReal.add_sub_cancel_left hfs]
```

Comment:
Can you add the `setIntegral` version as well?

Reply:
Done (now in `MeasureTheory.Integral.SetIntegral`). After a cursory search, I found only [one place](https://github.com/leanprover-community/mathlib4/blob/d419680bd7d6378744b29e71e3e161a996e82178/Mathlib/Probability/Martingale/Convergence.lean#L404) in Mathlib where `setIntegral_compl` could be usefully dropped in.

Reply:
Yeah, looks like a good idea to use it there. Mind opening a separate PR for `setIntegral_compl`/`setLintegral_compl` and the potential use you found?